app/lib/account_session.rb

```diff
@@ -51,13 +51,45 @@ def serialise
 
   def to_hash
     {
-      access_token: access_token,
-      refresh_token: refresh_token,
+      access_token: @access_token,
+      refresh_token: @refresh_token,
       level_of_authentication: level_of_authentication,
     }
   end
 
+  def get_remote_attributes(remote_attributes)
+    remote_attributes.index_with { |name| oidc_do :get_attribute, { attribute: name } }
+  end
+
+  def set_remote_attributes(remote_attributes)
+    return unless remote_attributes.any?
+
+    oidc_do :bulk_set_attributes, { attributes: remote_attributes }
+  end
+
+  def has_email_subscription?
+    oidc_do :has_email_subscription
+  end
+
+  def set_email_subscription(slug)
+    oidc_do :update_email_subscription, { slug: slug }
+  end
```

Comment:
It would be good to see these tested.

Reply:
They're covered by the request specs, do you think they also need tests just for this class?

Reply:
I think having them unit tested would be good because we rely on them in the controllers and they rely on a private method. It would also document them and make it clearer to see what they do and what they return.

Reply:
Ok, I've added tests in d25029373d8ef47b2c1e1972d7e0a64549740797

Reply:
Nice 👍